gcp_airflow_foundations/source_class/source.py

```diff
@@ -142,3 +137,76 @@ def default_task_args_for_table(
             "start_date": config.table_start_date(table_config),
         }
+
+    def create_dag(self, table_config: SourceTableConfig):
+        data_source = self.config.source
+        table_default_task_args = self.default_task_args_for_table(
+            self.config, table_config
+        )
+        logging.info(f"table_default_task_args {table_default_task_args}")
+
+        kwargs = data_source.dag_args if data_source.dag_args else {}
+
+        with DAG(
+            dag_id=f"{data_source.name}.{table_config.table_name}",
+            description=f"{data_source.name} to BigQuery load for {table_config.table_name}",
+            schedule_interval=data_source.ingest_schedule,
+            default_args=table_default_task_args,
+            catchup=data_source.catchup,
+            render_template_as_native_obj=True,
+            **kwargs,
+        ) as dag:
+
+            self.create_dag_tasks(dag, data_source, table_config)
+            return dag
+
+    def create_dag_source_level(self, template_config: SourceTemplateConfig, table_names):
+        data_source = self.config.source
+        table_default_task_args = self.default_task_args_for_table(
+            self.config, template_config
+        )
+        logging.info(f"table_default_task_args {table_default_task_args}")
+
+        kwargs = data_source.dag_args if data_source.dag_args else {}
+
+        with DAG(
+            dag_id=f"{data_source.name}.{template_config.template_ingestion_options.ingestion_name}",
+            description=f"{data_source.name} to BigQuery load for {template_config.template_ingestion_options.ingestion_name}",
+            schedule_interval=data_source.ingest_schedule,
+            default_args=table_default_task_args,
+            catchup=data_source.catchup,
+            render_template_as_native_obj=True,
+            **kwargs,
+        ) as dag:
+
+            for table in table_names:
+                template_config.table_name = table
+                template_config.landing_zone_table_name_override = \
+                    template_config.landing_zone_table_name_override_template.replace("{table}", table)
+                with TaskGroup(group_id=template_config.table_name) as table_task_group:
+                    self.create_dag_tasks(dag, data_source, template_config)
+                table_task_group
+
+            return dag
+
+    def create_dag_tasks(self, dag, data_source, table_config):
+        load_to_bq_landing = self.get_bq_ingestion_task(dag, table_config)
+        self.get_datastore_ingestion_task(
+            dag, load_to_bq_landing, data_source, table_config
+        )
+
+    def get_templated_table_list(self, templated_ingestion_options):
+        if templated_ingestion_options.ingest_mode.value == "INGEST_BY_TABLE_NAMES":
+            return templated_ingestion_options.table_names
+
+        full_table_list = self.get_source_tables_to_ingest()
+
+        if templated_ingestion_options.ingest_mode.value == "INGEST_ALL":
```

Comment:
I think you can replace it with TemplateIngestion.INGEST_ALL (instead of "INGEST_ALL")

Reply:
right, will do